### PR TITLE
[JENKINS-68031] Jenkins 2.0 style config form bottom border is cut off

### PR DESCRIPTION
**Pull request by frankie139506**
See [JENKINS-68031](https://issues.jenkins-ci.org/browse/JENKINS-68031).

圖片

### Proposed changelog entries

* Make bottom border visible in configure page (regression in 2.335).

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [X] (If applicable) Jira issue is well described
- [X] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [X] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by daniel-beck** [comment]
Interesting, this was deliberate? Looked very weird though, especially with the rounded corners.

File: war/src/main/js/widgets/config/tabbar.less
Comment (NotMyFault) [inline war/src/main/js/widgets/config/tabbar.less:112]: This part solely affects the buttons if they are at the end of the page:

https://user-images.githubusercontent.com/13383509/158563402-3e551ceb-b938-4992-8846-1bd9939ccf66.mp4

However, the buttons are sticky and move if you scroll up or down, hence the change proposed doesn't address the issue fully.
On previous versions, the buttons were honored in a separate box at the bottom left of the configuration tab:

Maybe we can reintroduce it looking more modern, thoughts?

**Comment by frankie139506** [comment]
what does `sticky and move` means?

File: war/src/main/js/widgets/config/tabbar.less
Comment (NotMyFault) [reply war/src/main/js/widgets/config/tabbar.less:112]: The bar sticks to the bottom of your screen and moves up and down depending on your scroll behavior.
The current change proposed solely takes care of the white bar at the bottom of the screen, it doesn't address the scrolling part.
I admit, the wording I've chosen was a bit misleading, apologies.

Comment (NotMyFault) [reply war/src/main/js/widgets/config/tabbar.less:112]: The bar sticks to the bottom of your screen and moves up and down depending on your scrolling behavior.
The current change proposed solely takes care of the bar if you are at the end of the configuration box, it's still visible at the scrolling part.
I admit, the wording chosen was a bit misleading, apologies.

Comment (frankie139506) [reply war/src/main/js/widgets/config/tabbar.less:112]: The upper border line cannot be seen when you scrolling at configuration box. If border-bottom is added when it is not scrolling to the bottom, it will be inconsistent up and down.

Comment (daniel-beck) [reply war/src/main/js/widgets/config/tabbar.less:112]: Isn't it the scrollspy tabs that provide the top border?

Comment (frankie139506) [reply war/src/main/js/widgets/config/tabbar.less:112]: 動畫
@daniel-beck The top border is invisible when you are not at the top.

**Comment by timja** [comment]
The page has been redesigned in https://github.com/jenkinsci/jenkins/pull/6485

so this is likely not needed from my understanding

**Comment by NotMyFault** [comment]
Yeah, Jan's PR supersedes it with a whole new, better design.

**Comment by basil** [comment]
True, the redesign of the **Configure Project** page in #6485 may solve [JENKINS-68031](https://issues.jenkins-ci.org/browse/JENKINS-68031), and if that PR is merged first, then this PR can certainly be closed. However, #6485 remains unmerged, so this PR is not without independent value. As a targeted bug fix for the regression described in JENKINS-68031, this PR has low risk (and is therefore safe for a backport), while the redesign of the **Configure Project** page in #6485 is high risk (and therefore not likely a good candidate for backport).

**Review by basil — APPROVED** [review]
The original bug as filed by @daniel-beck states:

> The bottom border of the `/job/xx/configure` form frame is cut off.

"Cut off" is a vague way of describing this, but I take this to mean the bottom border is missing. And this is indeed the case as of the current weekly, but not 2.332 (i.e., a minor regression). While this is not a major functional bug, restoring the missing bottom border fixes the small regression and gives the page the same look of polish that it once had.

As a targeted low-risk regression fix that restores the status quo prior to #5923, I think this PR makes sense. More sophisticated improvements to this page can be done in e.g. #6485, which is of larger scope, but pending the completion of #6485 I see no reason not to merge this in the short term.

**Comment by basil** [comment]
Is there a reason not to proceed with this PR to restore the status quo pending #6485? If there is such a reason, I am not seeing it yet.

**Comment by timja** [comment]
> Is there a reason not to proceed with this PR to restore the status quo pending #6485? If there is such a reason, I am not seeing it yet.

Just had higher priority issues to look at, checking it now

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
